pkg/combustion/osfiles.go

```diff
@@ -0,0 +1,89 @@
+package combustion
+
+import (
+	_ "embed"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/suse-edge/edge-image-builder/pkg/fileio"
+	"github.com/suse-edge/edge-image-builder/pkg/image"
+	"github.com/suse-edge/edge-image-builder/pkg/log"
+	"go.uber.org/zap"
+)
+
+const (
+	osFilesComponentName = "os files"
+	osFilesConfigDir     = "os-files"
+	osFilesScriptName    = "19-copy-os-files.sh"
+	osFilesLogFile       = "copy-os-files.log"
+)
+
+var (
+	//go:embed templates/19-copy-os-files.sh
+	osFilesScript string
+)
+
+func configureOSFiles(ctx *image.Context) ([]string, error) {
+	if !isComponentConfigured(ctx, osFilesConfigDir) {
+		log.AuditComponentSkipped(osFilesComponentName)
+		zap.S().Info("skipping os files component, no files provided")
+		return nil, nil
+	}
+
+	if err := copyOSFiles(ctx); err != nil {
+		log.AuditComponentFailed(osFilesComponentName)
+		return nil, err
+	}
+
+	if err := writeOSFilesScript(ctx); err != nil {
+		log.AuditComponentFailed(osFilesComponentName)
+		return nil, err
+	}
+
+	log.AuditComponentSuccessful(osFilesComponentName)
+	return []string{osFilesScriptName}, nil
+}
+
+func copyOSFiles(ctx *image.Context) error {
+	srcDirectory := filepath.Join(ctx.ImageConfigDir, osFilesConfigDir)
+	destDirectory := filepath.Join(ctx.CombustionDir, osFilesConfigDir)
+
+	dirEntries, err := os.ReadDir(srcDirectory)
+	if err != nil {
+		return fmt.Errorf("reading the os files directory at %s: %w", srcDirectory, err)
+	}
+
+	// If the directory exists but there's nothing in it, consider it an error case
+	if len(dirEntries) == 0 {
+		return fmt.Errorf("no files found in directory %s", srcDirectory)
+	}
+
+	logFilename := filepath.Join(ctx.BuildDir, osFilesLogFile)
+	logFile, err := os.Create(logFilename)
```

Comment:
This file is never used. Can we remove it?

Reply:
Yup, its a holdover from when I was shelling out to a script instead of using the Go implementation.